### ISSUE TITLE
Can't seem to get test ads to show up in iOS simulator

### ISSUE THREAD

**Issue by booboothefool** [callstack-io/react-native-fbads#45]
I used a placementId from the FAN dashboard and am running on a simulator, I'm getting nothing.

```
          import { BannerView } from 'react-native-fbads';

          <BannerView
            placementId="my_placement_id"
            type="large"
            onClick={() => console.log('click')}
            onError={(err) => console.log('error', err)}
          />
```

Am using React Native 0.37 with the latest version of this library 4.0.0. When I try using 3.1.1, I get the error: "Element type is invalid: expected a string (for built-in components) or a class/function for composite components) but got: undefined

What am I missing?

**Comment by booboothefool** [comment]
Oh what... 3.1.1 has no BannerView, and 4.0.0 isn't compatible with RN < 0.40?

**Comment by booboothefool** [comment]
On Android, I actually get an error that undefined is not an object (evaluating 'CTKNativeAdManager.init'). I ran `react-native link react-native-fbads`.

**Comment by booboothefool** [comment]
Got something to show up on my Android emulator. Just wasn't showing up on a real device for some reason. Still having issues with iOS. Adding FBAudienceNetwork.framework won't even build, so I tried it via CocoaPods.

**Comment by booboothefool** [comment]
I enabled some logging for iOS and am getting this:
```
[FBAudienceNetworkLog/FBAdProvider:146 thread:1 <error>] Ad request error: Error Domain=com.facebook.ads.sdk Code=2000 "Server Error" UserInfo={NSLocalizedDescription=Server Error, FBAdErrorDetailKey={
    code = 200;
    message = "(#200) Permissions error";
    type = OAuthException;
    url = "https://graph.facebook.com/network_ads_common/";
}}
```
Is it cause it doesn't realize it's a test device because the test device is not being properly added?

**Comment by booboothefool** [comment]
I got it!!!

This was crazy and way too difficult to set up. This was not documented anywhere. I had to read all over StackOverflow until I found this: http://stackoverflow.com/questions/33909430/how-to-test-facebook-audience-network-ads-over-testflight

I then added the snippets to my AppDelegate.m:
```
@import FBAudienceNetwork;
```
```
  NSLog(@"isTestMode1: %d",[FBAdSettings isTestMode]);
  NSString *testDeviceKey = [FBAdSettings testDeviceHash];
  if (testDeviceKey) {
    [FBAdSettings setLogLevel:FBAdLogLevelLog];
    [FBAdSettings addTestDevice:testDeviceKey];
  }
```

and voila!

![screen shot 2017-02-28 at 7 02 13 pm](https://cloud.githubusercontent.com/assets/13257645/23440174/747521b2-fde8-11e6-8903-79bdffdcefe1.png)

**What gives? Everyone else got this to work without doing all that nonsense?** 

Btw, running
```
React Native 0.37
FacebookSDK 4.18.0
react-native-fbads 3.1.1
```

**Comment by grabbou** [comment]
Great you managed to sort it out! Yes, we have been missing `addTestDevice` in both docs and APIs and recently figured out the reason it wasn't working.

Feel free to submit a PR fixing docs, I'll accept straight away!

**Comment by booboothefool** [comment]
@grabbou Cool! I saw your new release. So am I correct in that version > 4.0.0+ only works with React Native > 0.40? So if I'm stuck with React native 0.37, I need to stick with 3.1.1 of this library, where I just have to manually add the test device for now?

**Comment by grabbou** [comment]
Well, in that case I think you can keep it inside `AppDelegate.m` and mark as `to be removed when upgraded`.

**Comment by booboothefool** [comment]
@grabbou Cool, cool. 

Well my Android app is all set up and good to go. I am able to show test ads, and when I run it in production, I am able to show live ads. Clicked on one and Android app was just approved for Facebook Audience Network. 

On iOS, I was able to show the test ad with `[FBAdSettings addTestDevice:testDeviceKey];` in `AppDelegate.m`, however I'm not seeing a real ad after I remove it and run the app in Production/Profiling mode. For iOS, does my app actually have to be on the app store to display the ad I can click to send the approval request? I noticed in general, iOS requires a little more to get more ads most of the time, for example in my simulators where only real AdMob ads are shown on the Android simulator, but the iOS recognizes it as test: https://cloud.githubusercontent.com/assets/13257645/23438537/4acbfcf8-fde0-11e6-8e6e-147dc48fcfed.png

Any suggestions would be appreciated. Thanks. * so close *
